src/vsphere_cpi/spec/unit/cloud/vsphere/file_provider_spec.rb

```diff
@@ -2,86 +2,194 @@
 
 module VSphereCloud
   describe FileProvider, fake_logger: true do
-    subject(:file_provider) { described_class.new(http_client: http_client, vcenter_host: vcenter_host, retryer: retryer) }
+    subject(:file_provider) do  described_class.new(client: client,
+      http_client: http_client, vcenter_host: vcenter_host, retryer: retryer)
+    end
 
     let(:http_client) { double('fake-rest-client') }
     let(:vcenter_host) { 'fake-vcenter-host' }
     let(:retryer) { Retryer.new }
-
+    let(:client) { instance_double('VSphereCloud::VCenterClient') }
     let(:datacenter_name) { 'fake-datacenter-name 1' }
-    let(:datastore_name) { 'fake-datastore-name 1' }
     let(:path) { 'fake-path' }
-
+    let(:datastore) do instance_double('VimSdk::Vim::Datastore',
+      name: 'fake-datastore-name 1')
+    end
+    let(:host) {instance_double('VimSdk::Vim::HostSystem', name: 'host')}
+    let(:ticket) do instance_double('VimSdk::Vim::SessionManager::GenericServiceTicket',
+      id: 'ticket')
+    end
     before do
+      allow(file_provider).to receive(:get_healthy_host).with(anything).and_return(host)
+      allow(file_provider).to receive(:get_generic_service_ticket).with(anything).and_return(ticket)
       allow(retryer).to receive(:sleep)
     end
 
     describe '#fetch_file' do
-      it 'gets specified file' do
+      it 'first tries to get file from datastore via a host and succeeds' do
         response_body = double('response_body')
         response = double('response', code: 200, body: response_body)
+        expect(http_client).to receive(:get).with(
+          'https://host/folder/fake-path?'\
+          'dsName=fake-datastore-name%201',
+          {'Cookie' => "vmware_cgi_ticket=ticket"}
+        ).and_return(response)
+
+        expect(
+          file_provider.fetch_file_from_datastore(datacenter_name, datastore, path)
+        ).to eq(response_body)
+      end
+
+      it 'fails to get file via host and retries to get through datacenter and gets specified file' do
```

Comment:
'fails to get file via host and retries to get it through datacenter and succeeds '